spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/async/LazyTraceExecutor.java

```diff
@@ -47,7 +54,17 @@ public void execute(Runnable command) {
 				this.delegate.execute(command);
 			}
 		}
-		this.delegate.execute(new TraceRunnable(this.tracer, command));
+		if (this.traceKeys == null) {
+			try {
+				this.traceKeys = this.beanFactory.getBean(TraceKeys.class);
```

Comment:
Why is there no default? I'd expect it to be the other way around, assume the normalized keys and expect the occasional override